src/marginaldamage.jl

```diff
@@ -1,55 +1,88 @@
 """
-function compute_scc_mm(m::Model=get_model(); year::Union{Int, Nothing} = nothing, last_year::Int = model_years[end], prtp::Float64 = 0.015, eta::Float64=1.5)
+function compute_scc_mm(m::Model=get_model(); year::Union{Int, Nothing} = nothing, 
+    last_year::Int = model_years[end], prtp::Float64 = 0.015, eta::Float64=1.5, 
+    equity_weighting::Bool = false, normalization_region::Union{Int, Nothing} = nothing)
+
     year === nothing ? error("Must specify an emission year. Try `compute_scc_mm(m, year=2015)`.") : nothing
     !(last_year in model_years) ? error("Invlaid value of $last_year for last_year. last_year must be within the model's time index $model_years.") : nothing
     !(year in model_years[1]:10:last_year) ? error("Cannot compute the scc for year $year, year must be within the model's time index $(model_years[1]):10:$last_year.") : nothing
 
     mm = get_marginal_model(m; year = year)
-    scc = _compute_scc(mm; year=year, last_year=last_year, prtp=prtp, eta=eta)
+    scc = _compute_scc(mm; year=year, last_year=last_year, prtp=prtp, eta=eta, equity_weighting=equity_weighting, normalization_region=normalization_region)
     
     return (scc = scc, mm = mm)
 end
 
 # helper function for computing SCC from a MarginalModel, not to be exported or advertised to users
-function _compute_scc(mm::MarginalModel; year::Int, last_year::Int, prtp::Float64, eta::Float64)
+function _compute_scc(mm::MarginalModel; year::Int, last_year::Int, prtp::Float64, eta::Float64, equity_weighting::Bool = false, normalization_region::Union{Int, Nothing} = nothing)
     ntimesteps = findfirst(isequal(last_year), model_years)     # Will run through the timestep of the specified last_year 
     run(mm, ntimesteps=ntimesteps)
 
     marginal_damages = -1 * mm[:neteconomy, :C][1:ntimesteps, :] * 10.0^12 * 12/44 # convert from trillion $/ton C to $/ton CO2; multiply by -1 to get positive value for damages
-    global_marginal_damages = dropdims(sum(marginal_damages, dims = 2), dims=2)
 
-    global_c = dropdims(sum(mm.base[:neteconomy, :C], dims = 2), dims=2)
-    global_pop = dropdims(sum(mm.base[:neteconomy, :l], dims = 2), dims=2)
-    global_cpc = 1000 .* global_c ./ global_pop
+    # get consumption and population
+    c = mm.base[:neteconomy, :C]
+    pop = mm.base[:neteconomy, :l]
+    
+    num_regions = length(dim_keys(mm.base, :regions))
+
+    # get consumption per capita matrix
+    if equity_weighting
+        cpc = 1000 .* (c ./ pop)
+    else
+        cpc = 1000 .* repeat(sum(c, dims = 2) ./ sum(pop, dims = 2), 1, num_regions)
+    end
 
     year_index = findfirst(isequal(year), model_years)
+    
+    if isnothing(normalization_region)
+        cpc_0 = 1000 * sum(c[year_index,:]) / sum(pop[year_index, :])
+    else
+        cpc_0 = 1000 * c[year_index, normalization_region] / pop[year_index, normalization_region]
+    end
+
+    df = zeros(year_index-1, num_regions)
```

Comment:
Is this the correct size for `df`? If I added a pulse in 2010, `year_index` would be `6` I think? So now you would only have discount factors as a `5x12` array?

I think if you are only calculating discount factors for the periods after the pulse, you'd need something spanning from the year of pulse to the end of model time horizon. I'm not totally sure how you set up this internal SCC Mimi code. But in my own code, I would also just set discount factors to `0.0` for any periods before the pulse.

Reply:
In the deleted line 51 below, I think it essence were were starting with an array of zeros of length `year_index-1` and then concatenating that with one value calculated for each year. I think this mimics that with arrays, starting with a zeros array which would be `5x12` for a `year_index` of `6` and then appending a set of discount factors, one per region, for each of the following years between `year` and `last_year` inclusive.  I can probably do it more elegantly but does that make sense?

Reply:
Just for clarity the line I'm referring to is
```
df = [zeros(year_index-1)..., ((global_cpc[year_index]/global_cpc[i])^eta * 1/(1+prtp)^(t-year) for (i,t) in enumerate(model_years) if year<=t<=last_year)...]
```